src/domain/service/note.ts

```diff
@@ -11,28 +12,43 @@ export default class NoteService {
    */
   public repository: NoteRepository;
 
+  public noteRelationService: NoteRelationshipService;
+
   /**
    * Note service constructor
    *
    * @param repository - note repository
+   * @param noteRelationService - note relationship service
    */
-  constructor(repository: NoteRepository) {
+  constructor(repository: NoteRepository, noteRelationService: NoteRelationshipService) {
     this.repository = repository;
+    this.noteRelationService = noteRelationService;
   }
 
   /**
    * Adds note
    *
    * @param content - note content
    * @param creatorId - note creator
+   * @param parentPublicId - parent note if exist
    * @returns { Note } added note object
    */
-  public async addNote(content: JSON, creatorId: Note['creatorId']): Promise<Note> {
-    return await this.repository.addNote({
+  public async addNote(content: JSON, creatorId: Note['creatorId'], parentPublicId: Note['publicId'] | undefined): Promise<Note> {
+    const note = await this.repository.addNote({
       publicId: createPublicId(),
       content,
       creatorId,
     });
+
+    if (parentPublicId !== undefined) {
+      const parentNote = await this.getNoteByPublicId(parentPublicId);
+
+      if (parentNote !== null) {
+        await this.noteRelationService.addNoteRelation(note.id, parentNote.id);
+      }
+    }
```

Comment:
should we really make new note if invalid parentId passed?

Reply:
You mean we should throw exception if it is happend?

Reply:
maybe return null in service and in route return reply.notFound('parent note not found')